Replace clang-tidy job with general CI for Space ROS packages.

nuclearsandwich opened the pull request:
This job runs the full test suite for Space ROS packages minus ikos, which requires a custom build and is handled in a separate job.

nuclearsandwich committed:
Add general CI for Space ROS packages.

nuclearsandwich commented on rolling/ci-spaceros.yaml:
```diff
@@ -0,0 +1,98 @@
+  ROS_PYTHON_VERSION: '3'
+build_tool: colcon
+build_tool_args: '--cmake-args -DCMAKE_BUILD_TYPE=Debug -DEXPORT_COMPILE_COMMANDS=ON --no-warn-unused-cli'
+build_tool_test_args: '--retest-until-pass 2 --ctest-args -LE "(ikos|xfail)" --pytest-args -m "not xfail"'
```
The ikos tests are only run as ctest tests but I could add a pytest mark for consistency if that is preferred.

mjeronimo replied:
Seems like the consistency would be good. Does that mean that the IKOS tests would be run with either ctest or pytest? I'm not that familiar with the pytest marks.

nuclearsandwich replied:
Items passed to --ctest-args will get passed through to any package that has tests run via ctest (all CMake-derived packages) and likewise --pytest-args gets passed to any python packages (I don't remember the exact facilities but as far as I know pytest is the default test "runner" for colcon-python-setup-py packages.)

In our workflow ament_ikos is only ever invoked as a ctest test. Invoking it via pytest would require a package author to write a custom pytest test using it, mark that test with the ikos label, and then it's unclear what that would even do since ROS packages at least don't readily support Python packages with native extensions (CMake or ament_cmake is used in that case). So the only reason to add the label exclusion is to prevent people from asking "why is this _here_ but not _there_".

nuclearsandwich committed:
Remove dedicated clang-tidy build.

It is covered by spaceros CI.

nuclearsandwich commented:
Since this job runs ament_clang_tidy anyway I removed the dedicated clang-tidy builds to avoid double-covering.

nuclearsandwich committed:
Pad ctest regex argument with a space for shell quoting.

This is a temporay workaround to deal with some odd shell quoting
behavior in the build farm scripts stack. The test args substring
`(xfail|ikos)` is quoted since it contains shell control characters but
the quotes are being stripped from the string at some point when being
passed through the ros_buildfarm scripts (which could be happening in a
layer outside of our direct authorship) despite working as expected in a
neigboring substring `"not xfail and not ikos"`.

Scott astutely observed that the latter substring contains spaces while
the problematic regex does not. Padding the regex with an optional
trailing space, i.e. `(xfail|ikos) ?` is sufficient to prevent the
stripping of quotes from the substring and allow the job to run without
shell errors.

nuclearsandwich committed:
Update repos file url.

nuclearsandwich committed:
Use consistent(ly terrible) indentation convention.

nuclearsandwich committed:
Update ignored rosdep keys and packages.

nuclearsandwich committed:
Ignore both ament_cmake_ikos and ament_ikos.

nuclearsandwich committed:
Drop package selection args.

Ignoring these packages doesn't allow the build to proceed since
ament_cmake_ikos is required by ament_lint_common.

Instead we'll have to set up such that ikos is available but ikos
compiler wrappers are ignored.

clalancette commented:
We ended up deleting this job, so I don't think we are moving forward with this.  I'm going to close this out, but leave the branch around just in case.
